writer.go

```diff
@@ -102,12 +102,18 @@ func NewGenericWriter[T any](output io.Writer, options ...WriterOption) *Generic
 
 	if schema == nil && t != nil {
 		schema = schemaOf(dereference(t))
+		if len(schema.Columns()) == 0 {
+			panic("generic writer must be instantiated with type that has at least one exported field.")
+		}
 		config.Schema = schema
 	}
 
 	if config.Schema == nil {
 		panic("generic writer must be instantiated with schema or concrete type.")
 	}
+	if len(config.Schema.Columns()) == 0 {
+		panic("generic writer must be instantiated with schema that has at least one column.")
+	}
```

Comment:
I'm not sure about these checks, even if it's a weird condition, `parquet.GenericWriter[struct{}]` seems like a valid construct (the parquet spec allows empty schemas).

If not essential to this change, maybe we can discuss in a separate PR?

Reply:
It's a bit surprising that the file spec would allow zero columns... what does the file even contain in that case? If there are no columns then there can also be no rows and no data at all (since it's a columnar format).

When testing the earlier form of the PR (first commit), I had inadvertently created my test struct with all unexported fields (so no columns), and it then resulted in a panic only when closing the writer. So I had added these checks so that it would fail fast with a reasonable error message instead of `panic: runtime error: index out of range [0] with length 0`. 

Anyhow, I'll back it out here and open a separate PR where we can discuss further.

Reply:
Removed

Reply:
Thanks for sharing why you added the checks.

Your comment is fair, tho one could still create empty files with metadata (not sure why but who knows? 🤷)

I'll think more about it, seems like at least we'd add a test to validate whether we do or don't accept it, right now it's undefined behavior which isn't great.

Fixes for another day 👍